opm/autodiff/SimulatorBase_impl.hpp

```diff
@@ -381,6 +386,47 @@ namespace Opm
     }
 
     template <class Implementation>
+    void SimulatorBase<Implementation>::computeWellPotentials(const std::size_t step,
+                                                              const Wells* wells,
+                                                              const BlackoilState& x,
+                                                              const WellState& xw,
+                                                              std::vector<double>& well_potentials)
+    {
+        const int nw = wells->number_of_wells;
+        const int np = wells->number_of_phases;
+        well_potentials.clear();
+        well_potentials.resize(nw*np,0.0);
+        std::map<std::string, int> well_names_to_index;
+        if( ! xw.wellMap().empty() )
+        {
+            for (int w = 0; w < nw; ++w) {
+                for (int perf = wells->well_connpos[w]; perf < wells->well_connpos[w + 1]; ++perf) {
+                    const double well_cell_pressure = x.pressure()[wells->well_cells[perf]];
+                    const double drawdown_used = well_cell_pressure - xw.perfPress()[perf];
+                    const WellControls* ctrl = wells->ctrls[w];
+                    const int nwc = well_controls_get_num(ctrl);
+                    //Loop over all controls until we find a BHP control
+                    //that specifies what we need...
+                    double bhp = 0.0;
+                    for (int ctrl_index=0; ctrl_index < nwc; ++ctrl_index) {
+                        if (well_controls_iget_type(ctrl, ctrl_index) == BHP) {
+                            bhp = well_controls_iget_target(ctrl, ctrl_index);
+                        }
+                        // TODO: do something for thp;
+                    }
+                    const double drawdown_maximum = well_cell_pressure - (bhp + xw.well_perforation_pressure_diffs()[perf]);
```

Comment:
I wonder if we cannot avoid the well_perforation_pressure_diffs_ member entirely. Since we store xw.perfPress() already, and xw.bhp(), it can be recovered as the difference between them. That would simplify this PR a bit.